Add poster and video link

le-minhphuc opened the pull request:
Hi Prof Kan! 

I have add on the poster and video links - when you click the team's name you should be able to view a pop-up modal box containing video links and poster links

knmnyn commented:
Hi @le-minhphuc , you seem to have committed most of Bootstrap into the PR.  You do not need these files as BS is already incorporated into the project.  Suggest that you review some of the current CSS and HTML in the code so that you can reference the current installation and assets instead of adding them in again causing code replication.

Thanks!

knmnyn commented:
@le-minhphuc also you may want to check with the rest of the Engineering team as to how they handled the devise secret key problem.  To embed the secret keys is generally not a good idea.
Please do truncate the PR to ensure that the secret keys do not appear in the public.

Thanks!

le-minhphuc commented:
Dear Prof Kan,
Thank you for pointing out the devise secret key problem - I re-read the answers on the Internet and finally understand how to hide it using environment variable and gitignore. This website was very helpful: http://railsapps.github.io/rails-environment-variables.html. I have also fixed the problem and removed the duplicated bootstrap file. 
However I'm not sure why the travis build fail - the log says I'm using ruby 2.2.1 but my Gemfile write ruby 2.3.1. However, when I fixed it and try pushing to Heroku, the cmd line says I'm using ruby 2.3.1?

knmnyn commented:
Hi @le-minhphuc perhaps ask the rest of the Engineering team for help @njyjn @franklingu @ylhlh @wengkiat any ideas on Travis failures wrt to Ruby versions?  Please comment.

njyjn commented:
Phuc,

Your build report indicates an error here: 

> 326 Devise.secret_key was not set. Please add the following to your Devise initializer:

Can check if the ENV vars are properly set? Can refer to #472 for my implementation of a prev PR.

HNY!

le-minhphuc commented:
Thanks for your help Justin! I managed to resolve the devise secret problem in my fork by encrypting it in .travis.yml, but it still can't be resolved in nusskylab master repository though.

However, I'm now facing another problem. The SimpleCov gem and code-climate-reporter used in our project seems to have been deprecated, causing the `bundle exec rspec spec` step to throw error. Are you aware of this problem, and do you know how to address it? Thanks a lot

njyjn commented:
That's strange, didn't encounter this error before. Was an update to SimpleCov pushed recently or any file you may have modified?

le-minhphuc commented:
I first encountered the error after I fixed the ruby version error. The strange thing is when I check Gemfile.lock, our version of SimpleCov is still 0.10 (not 1.0 or above yet), and even when I run bundle update it is still 0.10. Maybe it's something on the TravisCI side.

Yes I've visited the change log and change things in the "Remove simplecov deprecated code" push and add code-climate-reporter as a separate step in the travis build, but it's still throwing the same error...

knmnyn commented:
Hi all:

Thanks for continuing to resolve the issue.  Later, please add both your
hours (inclusive of discussion and bugfixing) into the final comments on
the PR.


Cheers,

Min

--
Min-Yen KAN (Dr) :: Associate Professor :: National University of Singapore
:: NUS School of Computing, AS6 05-12, 13 Computing Drive
Singapore 117417 :: +65 6516 1885(DID) :: +65 6779 4580 (Fax) ::
kanmy@comp.nus.edu.sg (E) :: www.comp.nus.edu.sg/~kanmy (W)

On Sat, Dec 31, 2016 at 12:57 PM, le-minhphuc <notifications@github.com>
wrote:

> I first encountered the error after I fixed the ruby version error. The
> strange thing is when I check Gemfile.lock, our version of SimpleCov is
> still 0.10 (not 1.0 or above yet), and even when I run bundle update it is
> still 0.10. Maybe it's something on the TravisCI side.
>
> Yes I've visited the change log and change things in the "Remove simplecov
> deprecated code" push and add code-climate-reporter as a separate step in
> the travis build, but it's still throwing the same error...
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/nusskylab/nusskylab/pull/489#issuecomment-269849555>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AANP6z7NSfeArtuhH8e883INK3jwMLRpks5rNeCkgaJpZM4LX_yD>
> .
>

le-minhphuc commented:
Update: Oh yes your 2nd update seems to be what I'm looking for. Testing it now

njyjn commented:
@knmnyn noted, with thanks.

le-minhphuc commented:
Update: I think I found the answer at this link: https://docs.codeclimate.com/docs/travis-ci-ruby-test-coverage
However, to complete the code I would need the token supplied by CodeClimate for nusskylab, but since I'm not the owner of the repo I can't access it. @knmnyn Prof Kan, I understand that you would want to set this as an environment variable of the repo for security purpose. May I have the name of the environment variable to fill in my code? Also, I think the reason why Travis build step 1 succeed on my fork but not on master branch maybe because I name the environment variable differently compared to the repo's one. May I have the name of the environment var for devise secret token also?

knmnyn commented:
Hi, I think others have been able to fill in the code without needing extract CodeClimate env vars.  Can you check with the others first?

njyjn commented on config/initializers/devise.rb:
```diff
@@ -5,7 +5,6 @@
   # random tokens. Changing this key will render invalid all existing
   # confirmation, reset password and unlock tokens in the database.
   config.secret_key = ENV['DEVISE_SECRET_TOKEN']
```
Could this be the issue? Try using <%= ENV["DEVISE_SECRET_TOKEN"] %>

njyjn commented on test/test_helper.rb:
```diff
@@ -1,8 +1,8 @@
-require "codeclimate-test-reporter"
-CodeClimate::TestReporter.start
+require 'simplecov'
+SimpleCov.start
```
https://docs.codeclimate.com/docs/travis-ci-ruby-test-coverage advised to put this at 'the very top' of the file.

njyjn commented:
@le-minhphuc I just looked into it. We do have access to the repo env token, it is CODECLIMATE_REPO_TOKEN.

knmnyn commented:
For Devise it is an env var and called: "DEVISE_SECRET_TOKEN"=>"467 ...."
For Code Climate, running the rails c with production settings, I can't see the CODECLIMATE_REPO_TOKEN env var though.

knmnyn commented:
@le-minhphuc This doesn't actually open a pop-up modal box containing video links and poster links, it seems.
